src/Utilities/Compiler/Lightup/LightupHelpers.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.Lightup
+{
+    internal static class LightupHelpers
```

Comment:
Skipping review of light-up based code as I presume this was taken from StyleCop.

Reply:
It was taken _almost_ directly, and they are known to work well. The changes were:

1. Added support for converting to an equivalent enum type
2. Added support for generating a method with a 'symbol' parameter, as opposed to just 'syntax' (only affects the parameter name)
3. Added support for calling a getter method that takes an argument